modules/adtrgtmeBidAdapter.js

```diff
@@ -1,332 +1,358 @@
-    }
+  return oR;
+}
 
-    if (deepAccess(bid, 'ortb2Imp.ext.data') && isPlainObject(bid.ortb2Imp.ext.data)) {
-      impObject.ext.data = bid.ortb2Imp.ext.data;
-    };
+function appendImp(bid, oRtb) {
+  if (!oRtb || !bid) return;
+
+  const type = getType(bid);
+  const { floor: bidfloor = 0, currency: bidfloorcur = '' } = getBidfloor(bid);
+  const overrideFloor = deepAccess(bid, 'params.bidOverride.imp.bidfloor') || bidfloor;
+  const overrideCurrency = deepAccess(bid, 'params.bidOverride.imp.bidfloorcur') || bidfloorcur;
+
+  const impObject = {
+    id: bid.bidId,
+    secure: 1,
+    bidfloor: overrideFloor,
+    bidfloorcur: overrideCurrency,
+    ext: {
+      dfp_ad_unit_code: bid.adUnitCode,
+      ...(deepAccess(bid, 'ortb2Imp.ext.data') && isPlainObject(bid.ortb2Imp.ext.data) && { data: bid.ortb2Imp.ext.data })
+    },
+    ...(deepAccess(bid, 'params.zid') && { tagid: String(bid.params.zid) }),
+    ...(deepAccess(bid, 'ortb2Imp.instl') === 1 && { instl: 1 }),
+  };
 
-    if (deepAccess(bid, 'ortb2Imp.instl') && isNumber(bid.ortb2Imp.instl) && (bid.ortb2Imp.instl === 1)) {
-      impObject.instl = bid.ortb2Imp.instl;
+  if (type === BANNER) {
+    impObject.banner = {
+      mimes: bid.mediaTypes.banner.mimes || ['text/html', 'text/javascript', 'application/javascript', 'image/jpg'],
+      format: getFormat(bid.sizes),
+      ...(bid.mediaTypes.banner.pos && { pos: bid.mediaTypes.banner.pos }),
     };
+  }
 
-    openRtbObject.imp.push(impObject);
-  };
+  oRtb.imp.push(impObject);
 };
 
-function appendFirstPartyData(outBoundBidRequest, bid) {
-  const ortb2Object = bid.ortb2;
-  const siteObject = deepAccess(ortb2Object, 'site') || undefined;
-  const siteContentObject = deepAccess(siteObject, 'content') || undefined;
-  const userObject = deepAccess(ortb2Object, 'user') || undefined;
-
-  if (siteObject && isPlainObject(siteObject)) {
-    const allowedSiteStringKeys = ['name', 'domain', 'page', 'ref', 'keywords'];
-    const allowedSiteArrayKeys = ['cat', 'sectioncat', 'pagecat'];
-    const allowedSiteObjectKeys = ['ext'];
-    outBoundBidRequest.site = validateAppendObject(isStr, allowedSiteStringKeys, siteObject, outBoundBidRequest.site);
-    outBoundBidRequest.site = validateAppendObject(isArray, allowedSiteArrayKeys, siteObject, outBoundBidRequest.site);
-    outBoundBidRequest.site = validateAppendObject(isPlainObject, allowedSiteObjectKeys, siteObject, outBoundBidRequest.site);
-  };
-
-  if (siteContentObject && isPlainObject(siteContentObject)) {
-    const allowedContentStringKeys = ['id', 'title', 'language'];
-    const allowedContentArrayKeys = ['cat'];
-    outBoundBidRequest.site.content = validateAppendObject(isStr, allowedContentStringKeys, siteContentObject, outBoundBidRequest.site.content);
-    outBoundBidRequest.site.content = validateAppendObject(isArray, allowedContentArrayKeys, siteContentObject, outBoundBidRequest.site.content);
-  };
+function appendSiteData(oR, bid) {
```

Comment:
Why are you picking and choosing parts of the site object instead of taking the whole thing?

Reply:
Remaked it